integration/test/watcher/configmap/configmap_test.go

```diff
@@ -0,0 +1,355 @@
+// +build k8srequired
+
+package configmap
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/v2/pkg/label"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/v2/integration/key"
+	"github.com/giantswarm/app-operator/v2/pkg/annotation"
+	pkglabel "github.com/giantswarm/app-operator/v2/pkg/label"
+)
+
+// TestWatchingConfigMap tests app CRs are updated when wired configmaps are updated
+//
+// - Create user configmap, appcatalog configmap
+//
+// - Create app CR and wiring user configmap and appcatalog
+//
+// - Update user configmap and check the latest resource version is set on the annotation
+//   of app CR.
+//
+// - Update appcatalog onfigmap and check the latest resource version is set on the annotation
+//   of app CR.
+//
+//
+// - Delete app CR and check the watching label is deleted.
```

Comment:
Super nice and thanks for writing up the steps. Makes it easier to figure out what the test is doing!